source/common/stats/thread_local_store.cc

```diff
@@ -594,7 +594,12 @@ Gauge& ThreadLocalStoreImpl::ScopeImpl::gaugeFromStatNameWithTags(
   TagUtility::TagStatNameJoiner joiner(prefix_.statName(), name, stat_name_tags, symbolTable());
   StatName final_stat_name = joiner.nameWithTags();
 
-  StatsMatcher::FastResult fast_reject_result = parent_.fastRejects(final_stat_name);
+  StatsMatcher::FastResult fast_reject_result;
+  if (import_mode != Gauge::ImportMode::HiddenAccumulate) {
```

Comment:
Could you please add a comment here stating that we will always create hidden stats and ignore the stats_matcher unless we're rejecting all stats?

Reply:
It's currently implemented to ignore rejectAll if the stat is hidden. Is this what we would want considering deferred stats needs the hidden gauges?